folium/plugins/image_overlay.py

```diff
@@ -0,0 +1,134 @@
+# -*- coding: utf-8 -*-
+"""
+Image Overlay
+-------------
+
+Used to load and display a single image over specific bounds of
+the map, implements ILayer interface.
+
+"""
+import json
+from jinja2 import Template
+
+from folium.utilities import image_to_url
+from folium.map import Layer
+
+def mercator_transform(data, lat_bounds, origin='upper', height_out=None):
+    """Transforms an image computed in (longitude,latitude) coordinates into
+    the a Mercator projection image.
+
+    Parameters
+    ----------
+
+    data: numpy array or equivalent list-like object.
+        Must be NxM (mono), NxMx3 (RGB) or NxMx4 (RGBA)
+
+    lat_bounds : length 2 tuple
+        Minimal and maximal value of the latitude of the image.
+
+    origin : ['upper' | 'lower'], optional, default 'upper'
+        Place the [0,0] index of the array in the upper left or lower left
+        corner of the axes.
+
+    height_out : int, default None
+        The expected height of the output.
+        If None, the height of the input is used.
+    """
+    import numpy as np
+
+    mercator = lambda x: np.arcsinh(np.tan(x*np.pi/180.))*180./np.pi
+
+    array = np.atleast_3d(data).copy()
+    height, width, nblayers = array.shape
+
+    lat_min = max(lat_bounds[0], -85.051128779806589)
```

Comment:
This looks correct to me in this example: http://nbviewer.ipython.org/gist/ocefpaf/4e30b6b3d2426e6294d1

I do note that warping the image, both cartopy and the old function we had, looks "different" (not as smooth as the current version).  Not sure why and I haven't investigate what goes under the cover.

I will try to "beef up" the tests to check this and avoid future regressions.